CeedlingExec.py

```diff
@@ -0,0 +1,68 @@
+import os
+import sys
+
+import sublime
+import sublime_plugin
+from Default.exec import ExecCommand
+
+from .CeedlingSettings import CeedlingProjectSettings
+
+
+class CeedlingExecCommand(ExecCommand):
+    def run(self, **kwargs):
+
+        # "working_dir" is set by "new project" command.
+        #  project.xml does not exist unit project is created.
+        if kwargs.get("working_dir") is None:
+            try:
+                self.conf = CeedlingProjectSettings(self.window)
+
+            except OSError as e:
+                self.window.status_message("Ceedling: {}".format(e))
+                return
+
+            kwargs["working_dir"] = self.conf.working_dir
+
+        current_file = self.window.active_view().file_name()
+
+        if current_file is None:
+            current_file = current_file_name = ""
+
+        else:
+            _, current_ext = os.path.splitext(current_file)
+            if current_ext in ("", ".yml", ".rb"):
+                return
+
+            else:
+                current_file_name = os.path.basename(current_file)
+
+        task_sub = []
+
+        for t in kwargs.pop("tasks"):
+            for p, v in zip(
+                ["$file_name", "$file"],
+                [current_file_name, current_file],
+            ):
+                # verify tasks with a placeholder have a replacement value
+                if t.rfind(p) >= 0 and v == "":
+                    return
+
+                t = t.replace(p, v)
+
+            task_sub.append(t)
```

Comment:
```suggestion
        variables = self.window.extract_variables()
        task_sub = [sublime.expand_variables(task, variables) for task in kwargs.pop("tasks") or []]
```

Reply:
Thank you!  I knew there had to be a better way to do this.